pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix

```diff
@@ -0,0 +1,81 @@
+{
+  autoPatchelfHook,
+  fetchzip,
+  lib,
+  libfprint-tod,
+  openssl,
+  patchelfUnstable,
+  stdenv,
+}:
+
+# Based on ideas from (using a wrapper library to redirect fopen() calls to firmware files):
+#   * https://tapesoftware.net/replace-symbol/
+#   * https://github.com/NixOS/nixpkgs/pull/260715
+let
+  version = "5.12.018";
+  src = fetchzip {
+    url = "http://dell.archive.canonical.com/updates/pool/public/libf/libfprint-2-tod1-broadcom/libfprint-2-tod1-broadcom_${version}.orig.tar.gz";
+    hash = "sha256-0C2PpYpEJNrU+8NT95w4QV0J5nHQisMY94Czw3jQOzw=";
+  };
+
+  unpacked = stdenv.mkDerivation {
+    pname = "libfprint-2-tod1-broadcom-unpacked";
+    inherit src version;
+
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir $out
+      cp -r . $out
+    '';
+  };
+
+  # wraps `fopen()` for finding firmware files
+  wrapperLibName = "wrapper-lib.so";
+  wrapperLib = stdenv.mkDerivation {
+    pname = "libfprint-2-tod1-broadcom-wrapper-lib";
+    inherit version;
+
+    src = ./.;
+
+    postPatch = ''
+      substitute wrapper-lib.c lib.c \
+        --subst-var-by to "${unpacked}/var/lib/fprint/fw"
+      cc -fPIC -shared lib.c -o ${wrapperLibName}
+    '';
+
+    installPhase = ''
+      install -D -t $out/lib ${wrapperLibName}
+    '';
+  };
+in stdenv.mkDerivation {
+  pname = "libfprint-2-tod1-broadcom";
+  inherit src version;
+
+  buildInputs       = [ libfprint-tod openssl wrapperLib ];
+  nativeBuildInputs = [ autoPatchelfHook patchelfUnstable ];
```

Comment:
Done.

Reply:
I meant more like why is the default `patchelf` not sufficient – for example, if `--rename-dynamic-symbols` is not available on `patchelf`. This will help future maintainers to decide if it is fine to switch to regular `patchelf` when it is eventually updated.

Reply:
Ah, sure, makes sense. I added a corresponding comment (not in `nativeBuildInput` but  in the module's arguments at the file's top).